cdap-api/src/main/java/co/cask/cdap/api/flow/flowlet/AbstractFlowlet.java

```diff
@@ -106,6 +110,70 @@ protected void setProperties(Map<String, String> properties) {
   }
 
   /**
+   * @see FlowletConfigurer#addStream(Stream)
+   */
+  protected final void addStream(Stream stream) {
```

Comment:
this code shows up in a bunch of different places (also true for existing setName, setDescription, setProperties methods)? Is there a reason we don't have them all extend some common base class?